persistence/src/main/java/de/cyface/persistence/content/MeasurementProviderHelper.kt

```diff
@@ -0,0 +1,285 @@
+                        MeasurementTable.URI_PATH, LocationTable.URI_PATH, PressureTable.URI_PATH, EventTable.URI_PATH -> {
+                            val adaptedSelection = (BaseColumns.ID + "=" + rowIdentifier
+                                    + if (selection == null) "" else " AND $selection")
+                            ret += table.deleteRow(database, adaptedSelection, selectionArgs)
+                            database.setTransactionSuccessful()
+                            ret
+                        }
+                        else -> throw IllegalStateException("Unknown table identified by content provider URI: $uri")
+                    }
+                }
+                1 -> {
+                    when (pathSegments[0]) {
+                        // FIXME: ensure the measurement-related data is deleted automatically cascadingly (write/execute test)
+                        MeasurementTable.URI_PATH, LocationTable.URI_PATH, PressureTable.URI_PATH, EventTable.URI_PATH, IdentifierTable.URI_PATH -> {
+                            ret += table.deleteRow(database, selection, selectionArgs)
+                            database.setTransactionSuccessful()
+                            ret
+                        }
+                        else -> throw IllegalStateException(
+                            "Unable to delete data from table corresponding to URI $uri. There seems to be no such table."
+                        )
+                    }
+                }
+                else -> {
+                    throw IllegalStateException("Invalid path in content provider URI: $uri")
+                }
+            }
+        } finally {
+            database.endTransaction()
+        }
+    }
+
+    /**
+     * Inserts a single row into a table.
+     *
+     * @param uri The table to insert the row into.
+     * @param values The values to insert in the new table row.
+     * @return The identifier of the new row.
+     */
+    fun insertRow(uri: Uri, values: ContentValues): Long {
+        val table: CyfaceTable = matchTable(uri)
+        return table.insertRow(writableDatabase, values)
+    }
+
+    /**
+     * Inserts a list of `ContentValues` as new rows into a table.
+     *
+     * @param uri The table to insert the new rows into.
+     * @param values The values to insert.
+     * @return An array of identifiers for the newly created table rows.
+     */
+    fun bulkInsert(uri: Uri, values: List<ContentValues>): LongArray {
+        val table: CyfaceTable = matchTable(uri)
+        // FIXME: Why is insertBatch used here when we state everywhere that bulkInsert is 80x faster?
+        // The only old task I found is: https://cyface.atlassian.net/browse/MOV-248
+        // But nothing about bulkInsert. I guess we leave it like this as this is working
+        // writableDatabase.bulkInsert(uri, Arrays.asList(*values)).length
+        return table.insertBatch(writableDatabase, values)
+    }
```

Comment:
A layer above this method, we stated, that we noticed that `bulkInsert` is faster, but then we use `insertBatch` in the return call. I'm not sure why this is the case, but at least added documentation here, in case something gets slow again.

I didn't want to change this as there are no performance issues right now with the current SDK versions.

Also, as stated in a previous comment, we might be able to delete the content provider layer completely, anyways.